bindings/nodejs/lib/types/block/output/token-scheme.ts

```diff
@@ -26,39 +26,59 @@ abstract class TokenScheme {
  * Simple token scheme.
  */
 class SimpleTokenScheme extends TokenScheme {
-    private mintedTokens: HexEncodedAmount;
-    private meltedTokens: HexEncodedAmount;
-    private maximumSupply: HexEncodedAmount;
+    private mintedTokens: bigint;
+    private meltedTokens: bigint;
+    private maximumSupply: bigint;
 
     constructor(
-        mintedTokens: HexEncodedAmount,
-        meltedTokens: HexEncodedAmount,
-        maximumSupply: HexEncodedAmount,
+        mintedTokens: bigint,
+        meltedTokens: bigint,
+        maximumSupply: bigint,
     ) {
         super(TokenSchemeType.Simple);
-        this.mintedTokens = mintedTokens;
-        this.meltedTokens = meltedTokens;
-        this.maximumSupply = maximumSupply;
+        if (typeof mintedTokens === 'bigint') {
+            this.mintedTokens = mintedTokens;
+        } else if (mintedTokens) {
+            this.mintedTokens = hexToBigInt(mintedTokens);
+        } else {
+            this.mintedTokens = BigInt(0);
+        }
```

Comment:
out-of-interest (not a TS/JS guy): the `constructor` does not enforce `bigint` type already? So that's why an actual type check with `typeof` is still necessary and one could also pass in a String?

Reply:
If the bindings construct this object, it will be of type `string`, in runtime, the TS types aren't relevant, so if the user wants to construct this object, they can use `bigint`, but if it's constructed through the bindings, it's gonna be of type `string` and the `else if` clause will deal with that case. We could make it `bigint | string` but I think the bindings don't care about the type as I understood.

Reply:
okay